bin/utest/pgrp.c

```diff
@@ -45,6 +39,98 @@ static void sa_handler(int signo) {
   sig_delivered = 1;
 }
 
+int test_setpgid_leader(void) {
+  signal(SIGUSR1, sa_handler);
+
+  pid_t cpid = fork();
+  if (cpid == 0) {
+    /* Become session leader. */
+    assert(setsid() == getpid());
+    /* Can't change pgrp of session leader. */
+    assert(setpgid(0, 0));
+    while (!sig_delivered)
```

Comment:
Since we have `sigprocmask` and `sigsuspend` at our disposal, could you replace `sched_yield` with proper synchronisation?